File: casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class Casilla
    attr_accessor :numero_casilla
    attr_accessor :coste
    attr_accessor :tipo_casilla
    attr_accessor :titulo_propiedad
    
  public
  
  def const1(unNumero,unPrecio,unTipoCasilla,unTituloPropiedad)
    @numero_casilla=unNumero
    @coste=unPrecio
    @tipo_casilla=unTipoCasilla
    set_titulo(unTituloPropiedad)
  end
  
  def const2(unNumero,unTipoCasilla)
    @numero_casilla=unNumero
    @tipo_casilla=unTipoCasilla
    @coste=0
  end
  
  def initialize ()
    @numero=-1
    @coste=-1
  end
  
  def get_numero_casilla
    return @numero_casilla
  end
  
  def get_coste
    return @coste
  end
  
  def get_tipo_casilla
    return @tipo_casilla
  end
  
  def get_titulo_propiedad
    return @titulo_propiedad
  end
  
  @Overrride
  def to_s()
    if @coste==0 then
      return "Casilla: #{@numero_casilla} \n Tipo de Casilla: #{@tipo_casilla}" 
    else
      return "Casilla: #{@numero_casilla} \n Tipo de Casilla: #{@tipo_casilla} 
                \n Coste: #{@coste} \n Titulo Porpiedad: #{@titulo_propiedad}"
    end
 end
  
  private 
  def set_titulo(unTituloPropiedad)
    @titulo_propiedad=unTituloPropiedad
  end
  

end

File: QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author joseng2709
 */
public class Casilla {
   private int numeroCasilla;
   private int precioCompra;
   private TipoCasilla tipo;
   private TituloPropiedad titulo;
   
   public Casilla(TipoCasilla tipo,int numeroCasilla){
       this.tipo=tipo;
       this.numeroCasilla=numeroCasilla;
       precioCompra=0;
       this.titulo=new TituloPropiedad();
       
   }
   public Casilla(TipoCasilla tipo,int numeroCasilla,TituloPropiedad titulo){
       this.tipo=tipo;
       this.numeroCasilla=numeroCasilla;
       this.precioCompra=titulo.getPrecioCompra();
       setTitulo(titulo);
       
       
       
   }
   
   private void setTitulo (TituloPropiedad titulo){
       this.titulo=titulo;
   }
   public int getNumeroCasilla(){
       return numeroCasilla;
   }
   public int getPrecioCompra(){
       return precioCompra;
   }
   public TipoCasilla getTipo(){
       return tipo;
   }
   public TituloPropiedad getTituloPropiedad(){
       return titulo;
   }
   @Override
   public String toString(){
        // Tiene titulo, luego es una calle
        String comun = "Numero casilla = " + this.numeroCasilla + "\t" +
            "Coste = " + this.titulo.getPrecioCompra() + "\t" + 
            "Numero Hoteles = " + this.titulo.getNumHoteles() + "\t" +
            "Numero Casas = " + this.titulo.getNumCasas() + "\t" +
            "Tipo = " + this.tipo.toString() + "\n\n";
        if(this.titulo != null){
            return comun +
                "Titulo = " + this.titulo.toString() + "\n";
        }else{
            return comun;
        }
   }
}
